limitador-server/src/main.rs

```diff
@@ -258,6 +291,9 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     // structure needed to keep state of the last known canonical limits file path
     let mut last_known_canonical_path = fs::canonicalize(&limit_file).unwrap();
 
+    let status: Arc<RwLock<Status>> = Arc::default();
+    let status_updater = Arc::clone(&status);
```

Comment:
could you just use `status.clone()` ?
Not sure I follow this one...
Could create with Status::default() ?

Reply:
That's stylistic. They are the same. Usually the `Arc::clone` is preferred as it clearly indicate we are bumping a counter and not cloning the deref struct... in this case tho, yeah... it's obvious that `status` is an `Arc`

Reply:
Ugh... and yeah, that uses `Status::default()` actually, only implicitly. This `Arc::default()`, calls into `RwLock::default()`, which calls into `Status::default()` because of the explicit type on the lhs (`: Arc<RwLock<Status>>`). So we get a reference counted read/write lock protecting our `Status`. Syntactic sugar ftw :) ... not so for readability maybe?